### PR TITLE
[Tests] Added tests for poisson on periodic yaspgrid

### PR DESCRIPTION
**Pull request by linusseelinger**

### CODE REVIEW

**Comment by LSenf** [comment]
Btw, this requires the yaspgrid coordinate transformation to run successfully

File: dune/pdelab/test/CMakeLists.txt [hunk 1]
```diff
@@ -35,3 +35,9 @@ if(GMP_FOUND)
   pdelab_add_test(NAME testopbfem)
   pdelab_add_test(NAME test-dg-amg)
 endif(GMP_FOUND)
+
+# If we have MPI, include periodic tests
+if(MPI_FOUND)
+  pdelab_add_test(NAME testpoisson-periodic-2d COMMAND ${MPIEXEC} ${MPIEXEC_NUMPROC_FLAG} "4" testpoisson-periodic-2d)
```
Comment (smuething) [inline dune/pdelab/test/CMakeLists.txt]: You can use `pdelab_add_test(NAME testpoisson-periodic-2d MPIRANKS 4)` for this - the CMake macro will take care of the MPI dependency. That way, you can also remove the `MPI_FOUND` test

**Comment by LSenf** [comment]
Using pdelab_add_test now, added Q1/Q2 tests

**Comment by LSenf** [comment]
Rebased to catch up with master

**Pull request closed without merging**